### PR TITLE
JNG-5858 transform relation components

### PR DESCRIPTION
**Pull request by noherczeg**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-5858" title="JNG-5858" target="_blank">Sub-taskJNG-5858</a>  Transform relation visual elements
  </td></table>
  <br />

### CODE REVIEW

**Review by Madmat8 — APPROVED** [review]

**Review by gaborflorian — APPROVED** [review]

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl [hunk 1]
```diff
@@ -14,3 +14,114 @@ rule ViewLinkPageDefinition
...
+    to t: UI!ui::Column {
+        var firstStringFromTarget = s.referenceType.getAllPrimitiveFields().selectOne(f | f.referenceType.isDefined() and f.referenceType.`primitive` == "string");
+
+        if (firstStringFromTarget.isUndefined()) {
+            throw "Could not get string type field from relation target to use for representation: " + s.referenceType.name;
+        }
+
+        t.attributeType = firstStringFromTarget.getTransferFieldDeclarationEquivalent();
+        t.name = t.attributeType.name;
+        t.label = "";
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl:121]: why is label empty?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl [hunk 2]
```diff
@@ -14,3 +14,114 @@ rule ViewLinkPageDefinition
...
+
+        // TODO add action buttons
+
+        log.debug("InlineViewLinkButtonGroup: " + t.name);
+}
+
+@lazy
+rule ViewLinkDeclarationRepresentationColumn
+    transform s: JSL!ViewLinkDeclaration
+    to t: UI!ui::Column {
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl:110]: do columns not have id-s?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl [hunk 3]
```diff
@@ -14,3 +14,250 @@ rule ViewTablePageDefinition
...
+    transform s: JSL!ViewTableDeclaration
+    to t: UI!ui::Table
+    extends AbstractViewTableDeclaration {
+        guard: s.getTransferObjectType().isGenerated()
+
+        var id = actorDeclaration.name + "/(jsl/" + s.getId() + ")/InlineViewTable";
+        t.setId(id);
+        t.dataElement = s.equivalent("RelationType");
+
+        s.eContainer.uiContainer().children.add(t);
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl:60]: Is it really intended to modify the source object during the transformation?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl [hunk 4]
```diff
@@ -14,3 +14,250 @@ rule ViewTablePageDefinition
...
+        t.memberType = UI!ui::data::MemberType#DERIVED;
+        t.isReadOnly = true;
+        t.name = "_text_" + s.name;
+        t.isRequired = false;
+        t.isFilterable = true;
+
+        // TODO investigate if it really does not matter which string type we pick
+        t.dataType = actorDeclaration.equivalent("Application").dataTypes.selectOne(d | d.isTypeOf(UI!ui::data::StringType));
+
+        s.getContainerEquivalentClassType().attributes.add(t);
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl:235]: same concern as before (modifying source object)

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl [hunk 5]
```diff
@@ -14,3 +14,250 @@ rule ViewTablePageDefinition
...
+        log.debug("Created RowLinkDerivedAttributeType for ViewLinkDeclaration: [" + t.name + "] for [" + s.name + "]");
+}
+
+@lazy
+rule InlineViewTableButtonGroup
+    transform s: JSL!ViewTableDeclaration
+    to t: UI!ui::ButtonGroup {
+		t.setId(actorDeclaration.name + "/(jsl/" + s.getId() + ")/TableButtonGroup");
```
Comment (bencegelei) [inline judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl]: wrong id?

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl (hunk 3 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl:60]: `uiContainer()` returns transformed equivalent based on a set of parameters.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl (hunk 4 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewTableDeclaration.etl:235]: It doesn't do that.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl (hunk 1 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl:121]: It's not presented to users anywhere, only used for technical reasons.

File: judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl (hunk 2 above)
Comment (noherczeg) [reply judo-tatami-jsl-jsl2ui/src/main/epsilon/transformations/ui/modules/view/viewLinkDeclaration.etl:110]: fixed

**Pull request merged**